src/java.base/share/classes/java/io/DataOutputStream.java

```diff
@@ -30,7 +30,11 @@
 /**
  * A data output stream lets an application write primitive Java data
  * types to an output stream in a portable way. An application can
- * then use a data input stream to read the data back in.
+ * then use a data input stream to read the data back in. A data output
+ * stream wraps another output stream and delegates writing bytes to the
+ * write methods of that output stream. Hence writing data consisting of
+ * more than a single byte may cause several writes to the underlying
+ * output stream.
```

Comment:
This looks much better than the apiNote. One suggestion is to drop "Hence", I don't think it is needed here.